### ISSUE TITLE
ENH: gammainc to support not regularized lower incomplete gamma function.

### ISSUE THREAD

**Issue by JoeJoe1313** [scipy/scipy#18619]
### Describe the solution you'd like.

It would be nice to add an argument `regularized: bool` and have the possibility to choose between regularized calculation and not regularized.

**Comment by lucascolley** [comment]
@mdhaber you wrote a little about `gamma` and `regularized` in https://github.com/data-apis/array-api/issues/725. Any thoughts here?

**Comment by mdhaber** [comment]
+1. I imagine this is on the radar of @steppi and @izaid for the new library.

**Comment by izaid** [comment]
It's on the radar. https://github.com/scipy/scipy/pull/20539 adds normalised Legendre polynomials via a keyword argument, we'd use some of that to add a keyword argument for regularised gamma functions

**Comment by entaylor** [comment]
Thanks @mdhaber and @izaid for your efforts and your quick response on this!  I have continued to try to find potential workarounds, but have been stymied at every turn!  Adding this functionality will unlock my problem.

Not to be a pest, but just to try to manage my own projects, is it possible to know a rough timeline for this?

**Comment by mdhaber** [comment]
I'll let the others reply about timeline, but perhaps I can help in the meantime. Can you describe a bit more of what you need? What workarounds are not working? For example, why is numerical integration not working (if it's something you need immediately)? If it's accuracy, why not use `mpmath`? If speed is the main concern, do you also need high accuracy?

**Comment by entaylor** [comment]
Thanks, @mdhaber .  Yes, the first requirement is to get the unregularised incomplete gamma function, to allow us to consider a < 0.  As i say, this is possible with e.g. sympy or mpmath.  But the second requirement is to have something that we can use with pymc or numpyro, which seem (unless i'm mistaken?) to inherit the scipy gamma function definitions via jax.

I think our alternative is to try to code our own implementation of the unregularised incomplete gamma function using jax, but ... i'm a bit daunted by that prospect!

**Comment by dschmitz89** [comment]
Boost incudes lower and upper unregularized incomplete gamma functions as `tgamma_lower ` and `tgamma`: https://www.boost.org/doc/libs/latest/libs/math/doc/html/math_toolkit/sf_gamma/igamma.html . Wrapping boost functions is easy now in SciPy, so would adding those close this issue for you @entaylor @JoeJoe1313 ?

**Comment by entaylor** [comment]
Yes, i think that this would cover my use case.  Thank you for this!

**Comment by dschmitz89** [comment]
If you are interested in opening a PR yourself, you can mimic the recent PR that added the log of the regularized incomplete beta from boost: #25161 . :)
